### PR TITLE
[WIP] pre-trained, reduced fastText model

### LINKED ISSUE #8: see how pretrained fastText model can improve performance
**Is your feature request related to a problem? Please describe.**
The current fastText model has to learn language from scratch, which is difficult/impossible with little training data

**Describe the solution you'd like**
I would like there to be a method to use a pretrained fastText model, downloading either the generic Wikipedia model or creating an unsupervised pretrained model trained on github issues.

**Additional context**
Pretrained English model available at  https://fasttext.cc/docs/en/english-vectors.html

### LINKED ISSUE #67: implement pretrained W2V into fasttext models

### LINKED ISSUE #83: Reduce fastText models size
**Is your feature request related to a problem? Please describe.**
fastText has proven itself to be quite useful in predicting github labels, but the models are way too big, especially for the pre-trained version. It is not convenient to save multiple models and load them in the app. Some care has already been taken to reduce model size, such as reducing vocabulary and vector size in the src/data/build_w2v_vocab notebook. More can still be done to significantly reduce the size.

**Describe the solution you'd like**

Two main ideas are to quantize the model, which fastText allows easily in their python package. Another option is to throw out even more vocabulary (e.g. if a word doesn't appear once in our target issues), and reduce vector size yet again. We could have a finetuned vector model that is useful for anyone, and smaller, specific ones per user.

### LINKED ISSUE #93: make k-fold loop outside training and use same dataset for both models
**Is your feature request related to a problem? Please describe.**
Currently to test between fasttext and SVM we create the subdatasets twice and perform k-fold cross validation on them. Since we take negative samples randomly, this brings up some variance in the process. If fasttext and SVM are using two different datasets that have variance in their predictability, we cannot honestly compare them. 

**Describe the solution you'd like**
I would like the functions to be altered so both models use the same dataset. Instead of using 5-fold cross validation we should randomize the negative sampling 5 times, then train the models on 80% and average the 20% of the datasets where we do our validation.

### PR DESCRIPTION
**Pull request by antter**
Adding my fastText experiment notebook in a separate folder. This shows the results of using a pretrained ft model. Results are interesting to look through and show promise, but not enough to make it worth the hassle.

Fixes #67 
Fixes #8 

------------------------------------------------------------
NEW EDITS:

I reduced the model size substantially, so the vocab size is only the words that were actually used in the target repository and then performed PCA on it again.

Results were much better, outperforming untrained fastText substantially almost always, and outperforming SVMs a lot of the time. I also realized my validation loop isn't great, which is why I created issue #93 

Fixes #93 
Fixes #83

### CODE REVIEW

**Comment by antter** [comment]
@aakankshaduggal Cool if I close this? PR #95 has these features but actually implemented and not in an experimental directory

**Comment by aakankshaduggal** [comment]
okay go ahead with #95 instead then.
